roqoqo/tests/integration/operations/analog_operations.rs

```diff
@@ -0,0 +1,381 @@
+        &[
+            Token::Struct {
+                name: "ApplyConstantSpinHamiltonian",
+                len: 2,
+            },
+            Token::Str("hamiltonian"),
+            Token::Struct {
+                name: "SpinHamiltonianSerialize",
+                len: 2,
+            },
+            Token::Str("items"),
+            Token::Seq { len: Some(1) },
+            Token::Tuple { len: 2 },
+            Token::Str("0Z"),
+            Token::F64(1.0),
+            Token::TupleEnd,
+            Token::SeqEnd,
+            Token::Str("_struqture_version"),
+            Token::Struct {
+                name: "StruqtureVersionSerializable",
+                len: 2,
+            },
+            Token::Str("major_version"),
+            Token::U32(1),
+            Token::Str("minor_version"),
+            Token::U32(0),
+            Token::StructEnd,
+            Token::StructEnd,
+            Token::Str("time"),
+            Token::F64(1.0),
+            Token::StructEnd,
+        ],
+    );
+
+    assert_tokens(
+        &op.compact(),
+        &[
+            Token::Struct {
+                name: "ApplyConstantSpinHamiltonian",
+                len: 2,
+            },
+            Token::Str("hamiltonian"),
+            Token::Struct {
+                name: "SpinHamiltonianSerialize",
+                len: 2,
+            },
+            Token::Str("items"),
+            Token::Seq { len: Some(1) },
+            Token::Tuple { len: 2 },
+            Token::Seq { len: Some(1) },
+            Token::Tuple { len: 2 },
+            Token::U64(0),
+            Token::UnitVariant {
+                name: "SingleSpinOperator",
+                variant: "Z",
+            },
+            Token::TupleEnd,
+            Token::SeqEnd,
+            Token::NewtypeVariant {
+                name: "CalculatorFloat",
+                variant: "Float",
+            },
+            Token::F64(1.0),
+            Token::TupleEnd,
+            Token::SeqEnd,
+            Token::Str("_struqture_version"),
+            Token::Struct {
+                name: "StruqtureVersionSerializable",
+                len: 2,
+            },
+            Token::Str("major_version"),
+            Token::U32(1),
+            Token::Str("minor_version"),
+            Token::U32(0),
+            Token::StructEnd,
+            Token::StructEnd,
+            Token::Str("time"),
+            Token::NewtypeVariant {
+                name: "CalculatorFloat",
+                variant: "Float",
+            },
+            Token::F64(1.0),
+            Token::StructEnd,
+        ],
+    );
+}
+
+#[cfg(feature = "serialize")]
+#[test]
+fn timedependent_hamiltonian_serde() {
+    let pp = PauliProduct::new().z(0);
+    let mut hamiltonian = SpinHamiltonian::new();
+    hamiltonian
+        .add_operator_product(pp.clone(), CalculatorFloat::from(1.0))
+        .unwrap();
+
+    let mut values = HashMap::new();
+    values.insert("omega".to_string(), vec![1.0]);
```

Comment:
The above comment about this `omega` should be applied everywhere in this file, if relevant